### PR TITLE
Fix Collection#equals when nested Collections

### PR DESCRIPTION
**Pull request by Slamakans**
Tested a couple of times

### CODE REVIEW

File: src/util/ArraysEqual.js [hunk 1]
```diff
@@ -2,13 +2,5 @@ module.exports = function arraysEqual(a, b) {
...
-  for (const itemInd in a) {
-    const item = a[itemInd];
-    const ind = b.indexOf(item);
-    if (ind) {
-      b.splice(ind, 1);
-    }
-  }
-
-  return b.length === 0;
+  return a.every(e => b.includes(e));
```
Comment (Gawdl3y) [inline src/util/ArraysEqual.js]: This doesn't verify that the values are on the same indices. It's also outside of the scope of this PR.

File: src/util/Collection.js [hunk 2]
```diff
@@ -151,6 +153,7 @@ class Collection extends Map {
   find(propOrFn, value) {
     if (typeof propOrFn === 'string') {
       if (typeof value === 'undefined') throw new Error('Value must be specified.');
+      if (propOrFn === 'id') throw new RangeError('Don\'t use .find() with IDs. Instead, use .get(id).');
```
Comment (Gawdl3y) [inline src/util/Collection.js]: Why are you reintroducing this?

File: src/util/Collection.js [hunk 3]
```diff
@@ -357,9 +360,14 @@ class Collection extends Map {
     if (!collection) return false;
     if (this === collection) return true;
     if (this.size !== collection.size) return false;
-    return !this.find((value, key) => {
+    return this.every((value, key) => {
+      if (!collection.has(key)) return false;
       const testVal = collection.get(key);
-      return testVal !== value || (testVal === undefined && !collection.has(key));
+      try {
```
Comment (Gawdl3y) [inline src/util/Collection.js]: Objects/arrays should not be tested deeply - simply checking for basic reference equality was fine.

File: src/util/ArraysEqual.js (hunk 1 above)
Comment (Slamakans) [inline src/util/ArraysEqual.js]: Neither does the previous one, also the previous one wouldn't remove any item at index 0 (if-statement), but will run `b.splice(-1, 1)` when the item is not found.

Unless there's some black magic going on that I don't understand.

File: src/util/Collection.js (hunk 2 above)
Comment (Slamakans) [inline src/util/Collection.js]: It was an oops from copypasting

File: src/util/ArraysEqual.js (hunk 1 above)
Comment (Slamakans) [inline src/util/ArraysEqual.js]: If there is black magic going on, I would very much like to understand it :thinking:

File: src/util/Collection.js [hunk 4]
```diff
@@ -357,9 +357,10 @@ class Collection extends Map {
     if (!collection) return false;
     if (this === collection) return true;
     if (this.size !== collection.size) return false;
-    return !this.find((value, key) => {
+    return this.every((value, key) => {
+      if (!collection.has(key)) return false;
       const testVal = collection.get(key);
-      return testVal !== value || (testVal === undefined && !collection.has(key));
+      return testVal === value || (testVal && testVal.equals && value && value.equals && testVal.equals(value));
```
Comment (Gawdl3y) [inline src/util/Collection.js]: Checking for `value` and `value.equals` seems unnecessary, since you're not going to be calling those.

File: src/util/ArraysEqual.js (hunk 1 above)
Comment (Gawdl3y) [inline src/util/ArraysEqual.js]: Regardless, changing this is outside of the scope of this PR. Please make a new PR if you'd like to tweak `arraysEqual`.
